tests/integration/utils.py

```diff
@@ -148,9 +146,12 @@ def __get_object_type(object_path, types_dict, root_object_path=None):
             return obj['type']
         elif '$ref' in obj:
             # Avoid infinite recursion
-            if obj['$ref'] != root_object_path:
+            if root_object_paths is []:
+                root_object_paths.append(object_path)
+            if obj['$ref'] not in root_object_paths:
+                root_object_paths.append(obj['$ref'])
                 return __get_object_type(obj['$ref'], types_dict,
-                                         root_object_path)
+                                         root_object_paths)
         else:
             logging.warning('OpenAPI property contains no type or properties')
             return None
```

Comment:
I've changed the root object path to a collection of object paths that is appended each time __get_object_type is called. This is used to check for repeated recursion of any reference, not just the original.

If the reference is in the list of root paths, nothing is returned in that final condition. The warning is logged, and None returned.